sys/shell/commands/sc_netif.c

```diff
@@ -25,227 +27,343 @@
 #include "net/ng_netif.h"
 #include "net/ng_netapi.h"
 #include "net/ng_netconf.h"
+#include "net/ng_pkt.h"
 #include "net/ng_pktbuf.h"
 #include "net/ng_netif/hdr.h"
 
 /**
  * @brief   The maximal expected link layer address length in byte
  */
-#define MAX_ADDR_LEN            (16U)
-
+#define MAX_ADDR_LEN            (8U)
 
 /* utility functions */
-uint8_t _parse_addr(char *str, uint8_t *addr)
+static bool _is_number(char *str)
+{
+    for (; *str; str++) {
+        if (*str < '0' || *str > '9') {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+static bool _is_iface(kernel_pid_t dev)
+{
+    size_t numof;
+    kernel_pid_t *ifs = ng_netif_get(&numof);
+
+    for (size_t i = 0; i < numof; i++) {
+        if (ifs[i] == dev) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+static void _set_usage(char *cmd_name)
+{
+    printf("usage: %s <if_id> set <key> <value>\n", cmd_name);
+    puts("      Sets an hardware specific specific value\n"
+         "      <key> may be one of the following\n"
+         "       * \"addr\" - sets address\n"
```

Comment:
that would be a cyclic reference (since `addr_short` is documented as alias for `addr`). But I can change this line to `* \"addr\" - sets (short) address`